### ISSUE TITLE
QtValueError because of invalid URLs in _on_predicted_navigation

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3706]
When doing `dur` quickly (`:tab-close`, `:undo`, `:reload`):

```
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/qutebrowser/browser/webengine/webenginetab.py", line 934, in _on_predicted_navigation
    self.settings.update_for_url(url)
  File "/usr/lib/python3.6/site-packages/qutebrowser/config/websettings.py", line 149, in update_for_url
    value = values.get_for_url(url, fallback=False)
  File "/usr/lib/python3.6/site-packages/qutebrowser/config/configutils.py", line 160, in get_for_url
    if scoped.pattern is not None and scoped.pattern.matches(url):
  File "/usr/lib/python3.6/site-packages/qutebrowser/utils/urlmatch.py", line 278, in matches
    qtutils.ensure_valid(qurl)
  File "/usr/lib/python3.6/site-packages/qutebrowser/utils/qtutils.py", line 139, in ensure_valid
    raise QtValueError(obj)
qutebrowser.utils.qtutils.QtValueError: PyQt5.QtCore.QUrl('') is not valid
```

**Comment by The-Compiler** [comment]
Still seems to be an issue with some URLs on v1.2.1.

Reports:

- https://crashes.qutebrowser.org/view/3b51a840 (thanks @erikdsjostrom!)
- https://crashes.qutebrowser.org/view/1a8da039
- https://crashes.qutebrowser.org/view/75bef062 and others

URLs:

- https://lnkd.in/d2768Hc
- http://findfind.it and enter "http foo bar"

**Comment by The-Compiler** [comment]
This seems to be fixed, probably due to 7ac7ccc2967d19cae332e7082e380e7df27d1e75 and ee06ba0140ef5867abaae94b8270ea12960c10a9.
